mesonbuild/backend/ninjabackend.py

```diff
@@ -146,10 +146,16 @@ def ninja_quote(text: str, is_build_line: bool = False) -> str:
         raise MesonException(errmsg)
     return quote_re.sub(r'$\g<0>', text)
 
+
+@dataclass
 class TargetDependencyScannerInfo:
-    def __init__(self, private_dir: str, source2object: T.Dict[str, str]):
-        self.private_dir = private_dir
-        self.source2object = source2object
+
+    """Information passed to the depscanner about a target.
+    """
```

Comment:
Any particular reason why the ending triple-quotes is on a new line here for a one-line docstring?

Reply:
I thought that PEP257 said it should always go on a new line, but TIL that the closing `"""` only goes on its own line in a multi line docstring

Reply:
I'll just be not lazy and actually ad a full docstring